### PR TITLE
Add "canary rule" feature; drop match "type" from instrumentation; v1.5.0

### PR DESCRIPTION
**Pull request by mikeybtn**
See individual commits.

### CODE REVIEW

File: src/instrumenter.js [hunk 1]
```diff
@@ -91,13 +91,14 @@ class Instrumenter {
   /**
    * Record a HIT operation.
    * @param {string} status The status of the hit, either "accepted" or "rejected".
```
Comment (griffinmyers) [inline src/instrumenter.js]: Nit: update docstring for new statuses

File: tests/config-test.js [hunk 2]
```diff
@@ -204,10 +229,49 @@ describe('src/config', function () {
...
+
+      it('with a canary rule', function () {
+        const rules = config.findRules({
+          method: 'GET',
+          path: '/ping',
+          local: 'true',
+          ip: '1.2.3.4',
+        });
+        assert.equal(2, rules.length);
+        assert.deepEqual([
```
Comment (griffinmyers) [inline tests/config-test.js]: Recommend `assert.deepStrictEqual`: `assert.deepEqual` [has been deprecated](https://nodejs.org/api/assert.html#assert_assert_deepequal_actual_expected_message)... see old versions of node:

```node
> assert.deepEqual([], {})
undefined
> assert.deepEqual([], {wat: true})
AssertionError: [] deepEqual { wat: true }
    at repl:1:8
    at REPLServer.defaultEval (repl.js:248:27)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:412:12)
    at emitOne (events.js:82:20)
    at REPLServer.emit (events.js:169:7)
    at REPLServer.Interface._onLine (readline.js:210:10)
    at REPLServer.Interface._line (readline.js:549:8)
    at REPLServer.Interface._ttyWrite (readline.js:826:14)
```

Comment (griffinmyers) [reply tests/config-test.js]: Also legit to punt that to a future refactor PR, if you feel inclined.

File: tests/server-test.js [hunk 3]
```diff
@@ -186,6 +186,52 @@ describe('src/server', function () {
...
+        method: 'GET',
+        path: '/ping',
+        local: 'true',
+        ip: '1.2.3.4',
+      }).then(() => {
+        sinon.assert.callCount(instrumenter.countHit, 2);
+        sinon.assert.calledWith(instrumenter.countHit, 'canary-accepted', 'get-ping-by-ip-from-local');
+        sinon.assert.calledWith(instrumenter.countHit, 'accepted', 'get-ping-by-ip');
+        sinon.assert.callCount(instrumenter.timeHit, 1);
+        done();
```
Comment (griffinmyers) [inline tests/server-test.js]: If you're in a hurry you can just ditch the `done` ceremony and return the promise returned by `client.hit` to `it`.

File: README.md [hunk 4]
```diff
@@ -287,25 +395,32 @@ The server can be configured with several environment variables.
...
 
 * `divvy_tcp_connections_total`: Gauge of current open TCP connections.
 * `divvy_hit_duration_seconds`: Histogram of processing duration for HITs.
-* `divvy_hits_total`: Counter of all HIT operations.
-  * Labels: `status` (either `accepted` or `rejected`), `type` (either `none`, `rule`, or `default`).
-* `divvy_errors_total`: Counter of divvy errors.
-  * Labels: `code` (the type of error, such as `unknown-command`).
+* `divvy_hits_total`: Counter of all HIT operations. Labeled by:
+  * `status`: either `accepted` or `rejected` for normal rules; either `canary-accepted` or `canary-rejected` for rules with `"canary"` match policy.
+  * `rule_label`: the rule's `label` field, if specified. The value `default-reject` will be used for requests that match no rules.
```
Comment (mikeybtn) [inline README.md]:
```suggestion
  * `rule_label`: the rule's `label` field, if specified.
```

[the remark this answers is not in the document]
Comment (mikeybtn) [reply README.md]: you're right - stale comment!

**Pull request merged**